File: modules/expo-ble/src/ExpoBleModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import { ExpoBleModuleEvents } from './ExpoBle.types';

declare class ExpoBleModule extends NativeModule<ExpoBleModuleEvents> {
  setTheme: (theme: string) => void;
  getTheme: () => string;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<ExpoBleModule>('ExpoBle');

File: modules/expo-ble/android/src/main/java/expo/modules/ble/ExpoBleModule.kt
package expo.modules.ble

import android.content.Context
import android.content.SharedPreferences
import androidx.core.os.bundleOf
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoBleModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoBle")

    Function("setTheme") { theme: String ->
      getPreferences().edit().putString("theme", theme).commit()
    }

    Function("getTheme") {
      return@Function getPreferences().getString("theme", "system")
    }
  }

  private val context
  get() = requireNotNull(appContext.reactContext)

  private fun getPreferences(): SharedPreferences {
    return context.getSharedPreferences(context.packageName + ".settings", Context.MODE_PRIVATE)
  }
}

